torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1715,6 +1715,66 @@ std::vector<For*> LoopNest::reorder(
   return result;
 }
 
+void LoopNest::tile(For* x, For* y, int x_factor, int y_factor) {
+  auto parent = dynamic_cast<Block*>(x->get_parent());
+  if (parent == nullptr) {
+    throw malformed_input("parent of the loops must be a Block");
+  }
+  if ((x->body()->nstmts() != 1) || (x->body()->front() != y)) {
+    throw malformed_input("two loops must be perfectly nested");
+  }
+
+  // Split x, y axes by x_factor and y_factor
+  For *yo, *yi;
+  splitWithMask(y, y_factor, &yo, &yi);
+  For *xo, *xi;
+  splitWithMask(x, x_factor, &xo, &xi);
```

Comment:
What's the reason we're using `splitWithMask` instead of `splitWithTail`? Should it be configurable? I suspect on CPU `splitWithTail` would be better.